src/lib/guard/file-validator.guard.ts

```diff
@@ -0,0 +1,186 @@
+      signatures: [[0x7b], [0x5b]],
+    },
+    {
+      mime: 'application/zip',
+      signatures: [[0x50, 0x4b, 0x03, 0x04]],
+    },
+    {
+      mime: 'image/jpeg',
+      signatures: [
+        [0xff, 0xd8, 0xff, 0xe0],
+        [0xff, 0xd8, 0xff, 0xe1],
+        [0xff, 0xd8, 0xff, 0xe8],
+      ],
+    },
+    {
+      mime: 'image/png',
+      signatures: [[0x89, 0x50, 0x4e, 0x47, 0x0d, 0x0a, 0x1a, 0x0a]],
+    },
+  ];
+
+  constructor(private readonly allowedMimeTypes: string[] = []) {}
+
+  async canActivate(context: ExecutionContext): Promise<boolean> {
+    const request = context.switchToHttp().getRequest<FastifyRequest>();
+
+    if (!request.isMultipart()) {
+      throw new ForbiddenException('The request must be multipart');
+    }
+    await this.ensureUploadDirExists();
+
+    try {
+      const parts = request.parts();
+      const uploadedFiles: UploadedFile[] = [];
+
+      for await (const part of parts) {
+        if (part.type === 'file') {
+          const file = part as MultipartFile;
+          const uploadedFile = await this.saveFile(file);
+          const buffer = await fs.readFile(uploadedFile.path);
+
+          const isFile = await this.validateFile(buffer);
+          if (!isFile) {
+            if (this.allowedMimeTypes.indexOf('application/json') != -1) {
+              const isJson = await this.validateJson(buffer);
+              if (!isJson) {
+                fs.unlink(uploadedFile.path);
+                throw new ForbiddenException('Unauthorized file type');
+              }
+            }
+            throw new ForbiddenException('Unauthorized file type');
+          }
+
+          uploadedFiles.push(uploadedFile);
+        }
+      }
+
+      (request as any).uploadedFiles = uploadedFiles;
+
+      return true;
+    } catch (error) {
+      await this.cleanupFiles();
+
+      if (error instanceof ForbiddenException) {
+        throw error;
+      }
+      throw new ForbiddenException('Erreur lors du traitement du fichier');
+    }
+  }
+
+  private async saveFile(file: MultipartFile): Promise<UploadedFile> {
+    const buffer = await file.toBuffer();
+    const filename = file.filename;
+    const filepath = join(this.uploadDir, filename);
+
+    await fs.writeFile(filepath, buffer);
+
+    const originalSize = buffer.length;
+    const writtenSize = (await fs.stat(filepath)).size;
+
+    if (originalSize !== writtenSize) {
+      throw new Error(
+        `File integrity check failed: original size ${originalSize} != written size ${writtenSize}`
+      );
+    }
+
+    return {
+      originalName: file.filename,
+      filename: filename,
+      mimetype: file.mimetype,
+      size: buffer.length,
+      path: filepath,
+    };
+  }
+
+  private async ensureUploadDirExists(): Promise<void> {
+    try {
+      await fs.access(this.uploadDir);
+    } catch {
+      await fs.mkdir(this.uploadDir, { recursive: true });
+    }
+  }
+
+  private async cleanupFiles(): Promise<void> {
```

Comment:
The current cleanup strategy deletes all files which could be dangerous with concurrent uploads. Consider:

- Implementing a scheduled cleanup service using `@nestjs/schedule`
- Adding file tracking with metadata
- Setting configurable retention periods

Example:
```typescript
@Cron(CronExpression.EVERY_HOUR)
async cleanupStaleFiles() {
  // Remove files older than X hours
}
```